### ISSUE TITLE
`self_named_module_files` should ignore `build.rs`

### ISSUE THREAD

**Issue by jondo2010** [rust-lang/rust-clippy#9602]
### Summary

```
error: `mod.rs` files are required, found `my_crate/build.rs`
  |
  = note: requested on the command line with `-D clippy::self-named-module-files`
  = help: move `my_crate/build.rs` to `my_crate/build/mod.rs`
```

### Lint Name

self_named_module_files

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
<output>
```

I expected to see this happen:

**Comment by kraktus** [comment]
Can you share a repro tree file with the command used to reproduce the issue?
